socranop/installtool.py

```diff
@@ -51,6 +51,97 @@
 from socranop.dirs import get_dirs, init_dirs
 
 
+def print_step(tag, details, **printopts):
+    """Standardized output for all install/uninstall actions"""
+    print(f"  [{tag}] {details}", **printopts)
+
+
+class Step:
+
+    """\
+    Allow easy printing of checks similar to the following:
+
+        Some checks:
+          [load] module pydbus: OK
+          [load] module pydbus: ERROR
+        Error: Cannot import module 'pydbus'. Make sure the 'pydbus' package is installed.
+        Traceback (most recent call last):
+          File ...
+        ModuleNotFoundError: No module named 'pydbus'
+
+    by writing the following three liner:
+
+        with Step("load", "module pydbus",
+                  error_msg="Cannot import module 'pydbus'. Make sure the 'pydbus' package is installed."):
+            importlib.import_module("pydbus")  # import must work; discard retval
+    """
```

Comment:
This is awesome!

The only improvement I could imagine would be to have an optional 'exception_msgs' map that maps a specific exception type (such as `ModuleNotFoundError`) to a specific error message, with the logic on exit-for-exception being:
- If `exception_msgs[exc_type]` is set, print that
- If `error_msg` is set, print that
- If neither is set, use `str(exc_value)`

Reply:
We can easily do that if we ever need it. As of now, we have no use for it, so I would like to keep unexercised code out of the code base.